### PR TITLE
BFD-1931, BFD-1932 - remove unused beneficiary-related tables; restore parent claims FK constraints

### PR DESCRIPTION
**Pull request by mackec1**
**JIRA Ticket:**
[BFD-1931](https://jira.cms.gov/browse/BFD-1931)
[BFD-1932](https://jira.cms.gov/browse/BFD-1932)

**User Story or Bug Summary:**
Flyway migration scripts to remove old beneficiaries related tables and to re-constitute parent claims table foreign key (FK) constraints to _beneficiaries_new_ table.

### What Does This PR Do?

Drops the following tables which have been replaced by their __new_ version:
  - beneficiaries_history
  - medicare_beneficiaryid_history
  - beneficiary_monthly
  - skipped_rif_records
  - beneficiaries

Re-constitute foreign key constraints on _bene_id_ between parent claims tables and _beneficiaries_new_ table.
- adds FK _carrier_claims_bene_id_to_beneficiaries_ relationship between _carrier_claims_new_ and _beneficiaries_new_
- adds FK _dme_claims_bene_id_to_beneficiaries_ relationship between _dme_claims_new_ and _beneficiaries_new_
- adds FK _hha_claims_bene_id_to_beneficiaries_ relationship between _hha_claims_new_ and _beneficiaries_new_
- adds FK _hospice_claims_bene_id_to_beneficiaries_ relationship between _hospice_claims_new_ and _beneficiaries_new_
- adds FK _inpatient_claims_bene_id_to_beneficiaries_ relationship between _inpatient_claims_new_ and _beneficiaries_new_
- adds FK _outpatient_claims_bene_id_to_beneficiaries_ relationship between _outpatient_claims_new_ and _beneficiaries_new_
- adds FK _partd_events_bene_id_to_beneficiaries_ relationship between _partd_events_new_ and _beneficiaries_new_
- adds FK _snf_claims_bene_id_to_beneficiaries_ relationship between _snf_claims_new_ and _beneficiaries_new_

Create confluence page providing background on the approach taken for the ALTER TABLE, ADD CONSTRAINT which by definition requires an exclusive lock on the table; the standard approach of creating the FK and validating the cross-table data constraint is not feasible considering that we cannot afford to have a long-term, exclusive lock on tables being actively used by BFD services.

### What Should Reviewers Watch For?
* Verify that the flyway script(s) are well constructed, have comments, and perform the expected operation(s) with minimal impact on BFD services accessing the database.
* Verify that a Confluence page exists and conveys the strategy employed to execute the db operations for this PR.

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.
* Confluence page: https://confluence.cms.gov/display/BFD/Implementing+Foreign+Key+Constraints+in-place

### What Security Implications Does This PR Have?

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

* N/A

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V87__new_schema_rebuild_claims_fk_constraints.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# V85__rebuild_claims_fk_constraints
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V87__new_schema_rebuild_claims_fk_constraints.conf:1]: name and number are a little off now

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V87__new_schema_rebuild_claims_fk_constraints.sql [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+-- in 'NOT VALID' mode which allows the constraint to created but not validated,
+-- resulting in a very fast meta-data only change.
+--
+-- A side note for reviewers; the BFD standard naming convention calls for
+-- including the table name(s) as part of constraint name. We deviate slightly
+-- from this for the '_new' tables since:
+--   a) we are not currently using the original constraint name(s)
+--   b) by naming the constraints to use its original name, we eliminate
+--      the necessity to alter the constraint names once we rename all the
+--      _new tables back to their original name(s).
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V87__new_schema_rebuild_claims_fk_constraints.sql:40]: 👍

**Review by christrotteradhoc — COMMENTED** [review]
looks good ; very minor comments

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V87__new_schema_rebuild_claims_fk_constraints.sql [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+${logic.psql-only} alter table if exists public.hospice_claims_new 
+${logic.psql-only} validate constraint hospice_claims_bene_id_to_beneficiaries;
+
+${logic.psql-only} alter table if exists public.inpatient_claims_new 
+${logic.psql-only} validate constraint inpatient_claims_bene_id_to_beneficiaries;
+
+${logic.psql-only} alter table if exists public.outpatient_claims_new 
+${logic.psql-only} validate constraint outpatient_claims_bene_id_to_beneficiaries;
+
+${logic.psql-only} alter table if exists public.partd_Events_new
```
[the remark this answers is not in the document]
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V87__new_schema_rebuild_claims_fk_constraints.sql:145]: found it! fixed (although as you said, case does not matter in SQL unless you explicitly wrap in quotes)

**Review by christrotteradhoc — APPROVED** [review]
Nice work!

**Pull request merged**